planner/memo/group.go

```diff
@@ -87,15 +88,40 @@ type Group struct {
 	EngineType EngineType
 }
 
+// buildLogicalProperty builds a new LogicalProperty for the new GroupExpr.
+// it also builds the unique key info and max one row info.
+func buildLogicalProperty(e *GroupExpr, oldSchema *expression.Schema) *property.LogicalProperty {
+	prop := &property.LogicalProperty{}
+	if oldSchema == nil {
+		return prop
+	}
+	newSchema := expression.NewSchema(oldSchema.Columns...)
```

Comment:
Can we just modify `oldSchema`?

Reply:
In some rules, oldSchema is from it's child Group. If we use the oldSchema here, we may influence the schema(key info) of child Group.